### PR TITLE
Add specific http configurations for cloudwatch and s3

### PR DESCRIPTION
**Pull request by zoewangg**
## Description
- add cloudwatch specific http configurations, specifically reducing `connectionMaxIdleTime`
- Add s3 specific http configurations, specifically reducing `connectionMaxIdleTime`

## Motivation and Context
#1380 , #1122 

## Testing
Added low TPS long interval stability tests. Verified the tests failed w/o the changes and succeeded w the change.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)

## Checklist

- [x] I have read the **CONTRIBUTING** document
- [x] Local run of `mvn install` succeeds
- [x] My code follows the code style of this project
- [ ] My change requires a change to the Javadoc documentation
- [ ] I have updated the Javadoc documentation accordingly
- [x] I have read the **README** document
- [x] I have added tests to cover my changes
- [x] All new and existing tests passed
- [x] A short description of the change has been added to the **CHANGELOG**
- [ ] My change is to implement 1.11 parity feature and I have updated [LaunchChangelog](https://github.com/aws/aws-sdk-java-v2/blob/master/docs/LaunchChangelog.md)

## License

- [x] I confirm that this pull request can be released under the Apache 2 license

### CODE REVIEW

File: test/stability-tests/src/it/java/software/amazon/awssdk/stability/tests/cloudwatch/CloudWatchBaseStabilityTest.java [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+import software.amazon.awssdk.services.cloudwatch.CloudWatchAsyncClient;
+import software.amazon.awssdk.testutils.service.AwsTestBase;
+
+public abstract class CloudWatchBaseStabilityTest extends AwsTestBase {
+    protected static final int CONCURRENCY = 50;
+    protected static final int TOTAL_RUNS = 3;
+
+    protected static CloudWatchAsyncClient cloudWatchAsyncClient =
+        CloudWatchAsyncClient.builder()
+                             .httpClientBuilder(NettyNioAsyncHttpClient.builder().maxConcurrency(CONCURRENCY))
```
Comment (spfink) [inline test/stability-tests/src/it/java/software/amazon/awssdk/stability/tests/cloudwatch/CloudWatchBaseStabilityTest.java]: I forget how this works...does this one merge the custom config?

File: test/stability-tests/src/it/java/software/amazon/awssdk/stability/tests/TestRunner.java [hunk 2]
```diff
@@ -21,12 +21,12 @@
...
         // You can add the tests you want to run here.
         // eg:
-//        try {
-//            S3AsyncStabilityTest s3AsyncStabilityTest = new S3AsyncStabilityTest();
-//            S3AsyncStabilityTest.setup();
-//            s3AsyncStabilityTest.putObject_getObject();
-//        } finally {
-//            S3AsyncStabilityTest.cleanup();
-//        }
+        //        try {
```
Comment (spfink) [inline test/stability-tests/src/it/java/software/amazon/awssdk/stability/tests/TestRunner.java]: ?

File: test/stability-tests/src/it/java/software/amazon/awssdk/stability/tests/cloudwatch/CloudWatchBaseStabilityTest.java (hunk 1 above)
Comment (zoewangg) [reply test/stability-tests/src/it/java/software/amazon/awssdk/stability/tests/cloudwatch/CloudWatchBaseStabilityTest.java]: Yup

File: test/stability-tests/src/it/java/software/amazon/awssdk/stability/tests/TestRunner.java (hunk 2 above)
Comment (zoewangg) [reply test/stability-tests/src/it/java/software/amazon/awssdk/stability/tests/TestRunner.java]: Those are for testing purpose. I'll move the code snippets to javadoc section.

**Pull request merged**